### ***ydata-profiling checkpoint 1***

Objective
In this checkpoint, we are going to work on 'Education Attainment And Enrolment Around The World' dataset, provided by the World bank group.

Dataset description : The dataset was derived from the original copy, which was published on the world bank website. The primary purpose of this database is to document and analyse patterns of educational attainment rates in different country around the world.
###

***Instructions***
- Step 1: Data exploration with Pandas
- Download the provided dataset
- Use pandas to read the file into a pandas Dataframe.
- Display general information about the dataset.
- Run descriptive analysis on the dataset using the 'describe' function.
- Step 2: Data exploration with ydata-profiling
- Use ydata-profiling to generate a report of the provided dataset.
- Look for missing values in the dataset. You can use the report generated by ydata to identify missing values.
- Examine the correlations between different columns in the dataset. You can use the report generated by ydata to identify correlations.
Identify any other interesting patterns or insights in the dataset that you can discover through the ydata-profiling report.
- Step 3: Summary
- Write a summary of your findings based on the ydata-profiling report. Be sure to include any interesting patterns or insights you discovered in the dataset.

In [2]:
import pandas as pd
import seaborn as sns
from ydata_profiling import ProfileReport

In [3]:
df = pd.read_csv(r"C:\Users\DELL\Downloads\EDUCATION_ATTAINMENT.csv")

In [4]:
data_profiling = df.head(10)
data_profiling

Ages[15-49]_All_grade1  Ages[15-49]_All_grade2  Ages[15-49]_All_grade3  \
0                0.669091                0.667214                0.651812   
1                0.485879                0.480171                0.465510   
2                0.516431                0.514233                0.502269   
3                0.914294                0.903528                0.871690   
4                0.876990                0.841497                0.762864   
5                0.889589                0.852069                0.771968   
6                0.803733                0.746789                0.631720   
7                0.981062                0.981062                0.980476   
8                0.991079                0.985556                0.981743   
9                0.992481                0.990077                0.984427   

   Ages[20-29]_All_grade1  Ages[20-29]_All_grade2  Ages[20-29]_All_grade3  \
0                0.460899                0.458749                0.446925   
1                0.267236                0.265735                0.257999   
2                0.327775                0.326250                0.318003   
3                0.853770                0.840792                0.810794   
4                0.881265                0.854158                0.795892   
5                0.831665                0.794070                0.734560   
6                0.750124                0.720764                0.654041   
7                0.982854                0.982854                0.980559   
8                0.984294                0.980344                0.977792   
9                0.994806                0.994806                0.993977   

   Ages[30-39]_All_grade1  Ages[30-39]_All_grade2  Ages[30-39]_All_grade3  \
0                0.292791                0.290929                0.279100   
1                0.191279                0.188839                0.182693   
2                0.221981                0.220472                0.214040   
3                0.775725                0.757109                0.713569   
4                0.883625                0.853812                0.805231   
5                0.784236                0.737433                0.667450   
6                0.765275                0.737825                0.675347   
7                0.984600                0.984600                0.984600   
8                0.976801                0.971111                0.969247   
9                0.994772                0.994567                0.991585   

   Ages[40-49]_All_grade1  ...  Ages[20-29]_Female_grade2  \
0                0.250981  ...                   0.279915   
1                0.187151  ...                   0.122435   
2                0.205611  ...                   0.157744   
3                0.786468  ...                   0.786506   
4                0.747516  ...                   0.793379   
5                0.766246  ...                   0.707985   
6                0.637211  ...                   0.624397   
7                0.984417  ...                   0.987454   
8                0.989354  ...                   0.976583   
9                0.993558  ...                   1.000000   

   Ages[20-29]_Female_grade3  Ages[30-39]_Female_grade1  \
0                   0.272185                   0.111428   
1                   0.117916                   0.084555   
2                   0.151130                   0.086072   
3                   0.746073                   0.699617   
4                   0.712726                   0.808248   
5                   0.633392                   0.695446   
6                   0.546101                   0.649015   
7                   0.986111                   0.990672   
8                   0.972163                   0.974779   
9                   0.999269                   0.998587   

   Ages[30-39]_Female_grade2  Ages[30-39]_Female_grade3  \
0                   0.109101                   0.103791   
1                  

In [5]:
#Display general information about the dataset.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 626 entries, 0 to 625
Data columns (total 38 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Ages[15-49]_All_grade1     626 non-null    float64
 1   Ages[15-49]_All_grade2     626 non-null    float64
 2   Ages[15-49]_All_grade3     626 non-null    float64
 3   Ages[20-29]_All_grade1     626 non-null    float64
 4   Ages[20-29]_All_grade2     626 non-null    float64
 5   Ages[20-29]_All_grade3     626 non-null    float64
 6   Ages[30-39]_All_grade1     625 non-null    float64
 7   Ages[30-39]_All_grade2     625 non-null    float64
 8   Ages[30-39]_All_grade3     625 non-null    float64
 9   Ages[40-49]_All_grade1     625 non-null    float64
 10  Ages[40-49]_All_grade2     625 non-null    float64
 11  Ages[40-49]_All_grade3     625 non-null    float64
 12  Ages[15-49]_Male_grade1    626 non-null    float64
 13  Ages[15-49]_Male_grade2    626 non-null    float64

In [6]:
#Run descriptive analysis on the dataset using the 'describe' function.
df.describe()

Ages[15-49]_All_grade1  Ages[15-49]_All_grade2  Ages[15-49]_All_grade3  \
count              626.000000              626.000000              626.000000   
mean                 0.873886                0.861490                0.837509   
std                  0.162624                0.166895                0.176555   
min                  0.251917                0.244071                0.233274   
25%                  0.831791                0.803131                0.757465   
50%                  0.948926                0.935167                0.909587   
75%                  0.984447                0.980112                0.971166   
max                  1.000000                1.000000                1.000000   

       Ages[20-29]_All_grade1  Ages[20-29]_All_grade2  Ages[20-29]_All_grade3  \
count              626.000000              626.000000              626.000000   
mean                 0.823849                0.812738                0.789965   
std                  0.200559                0.202587                0.207979   
min                  0.079610                0.079610                0.079610   
25%                  0.744915                0.723418                0.678869   
50%                  0.910132                0.896568                0.864849   
75%                  0.973777                0.967410                0.955625   
max                  1.000000                0.999824                0.999264   

       Ages[30-39]_All_grade1  Ages[30-39]_All_grade2  Ages[30-39]_All_grade3  \
count              625.000000              625.000000              625.000000   
mean                 0.766171                0.752519                0.724980   
std                  0.236563                0.237679                0.240681   
min                  0.098059                0.091358                0.085236   
25%                  0.639447                0.612464                0.561545   
50%                  0.853861                0.831246                0.794898   
75%                  0.959097                0.948259                0.922841   
max                  1.000000                1.000000                1.000000   

       Ages[40-49]_All_grade1  ...  Ages[20-29]_Female_grade1  \
count              625.000000  ...                 626.000000   
mean                 0.709166  ...                   0.790544   
std                  0.259998  ...                   0.236541   
min                  0.036549  ...                   0.029662   
25%                  0.527055  ...                   0.678596   
50%                  0.785356  ...                   0.900869   
75%                  0.937129  ...                   0.974514   
max                  1.000000  ...                   1.000000   

       Ages[20-29]_Female_grade2  Ages[20-29]_Female_grade3  \
count                 626.000000                 626.000000   
mean                    0.778972                   0.755644   
std                     0.238637                   0.243755   
min                     0.029662                   0.029662   
25%                     0.658977                   0.616380   
50%                     0.887206                   0.853414   
75%                     0.968962                   0.959495   
max                     1.000000                   0.999777   

       Ages[30-39]_Female_grade1  Ages[30-39]_Female_grade2  \
count                 625.000000                 625.000000   
mean                    0.719020                   0.704342   
std                     0.278133                   0.279059   
min                     0.060930                   0.058368   
25%                     0.534060                   0.508218   
50%                     0.815823                   0.794049   
75%                     0.958022                   0.946510   
max                     1.000000                   1.000000   

       Ages[30-39]_Female_grade3  Ages[40-49]_Female_grade1  \
count                 625.000000

In [8]:
profile = ProfileReport(df, title='Education Attainment Profile', explorative=True)
profile.to_file("Education_Attainment_profile_report.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

C:\Users\DELL\anaconda3\Lib\site-packages\seaborn\matrix.py:260: FutureWarning: Format strings passed to MaskedConstant are ignored, but in future may error or produce different behavior
  annotation = ("{:" + self.fmt + "}").format(val)
C:\Users\DELL\anaconda3\Lib\site-packages\ydata_profiling\model\missing.py:78: UserWarning: There was an attempt to generate the Heatmap missing values diagrams, but this failed.
To hide this warning, disable the calculation
(using `df.profile_report(missing_diagrams={"Heatmap": False}`)
If this is problematic for your use case, please report this as an issue:
https://github.com/ydataai/ydata-profiling/issues
(include the error message: 'could not convert string to float: '--'')
  warnings.warn(


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [7]:
#Look for missing values in the dataset. You can use the report generated by ydata to identify missing values.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 626 entries, 0 to 625
Data columns (total 38 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Ages[15-49]_All_grade1     626 non-null    float64
 1   Ages[15-49]_All_grade2     626 non-null    float64
 2   Ages[15-49]_All_grade3     626 non-null    float64
 3   Ages[20-29]_All_grade1     626 non-null    float64
 4   Ages[20-29]_All_grade2     626 non-null    float64
 5   Ages[20-29]_All_grade3     626 non-null    float64
 6   Ages[30-39]_All_grade1     625 non-null    float64
 7   Ages[30-39]_All_grade2     625 non-null    float64
 8   Ages[30-39]_All_grade3     625 non-null    float64
 9   Ages[40-49]_All_grade1     625 non-null    float64
 10  Ages[40-49]_All_grade2     625 non-null    float64
 11  Ages[40-49]_All_grade3     625 non-null    float64
 12  Ages[15-49]_Male_grade1    626 non-null    float64
 13  Ages[15-49]_Male_grade2    626 non-null    float64

In [11]:
missing_values = print(df.isnull().sum())
missing_values

Ages[15-49]_All_grade1       0
Ages[15-49]_All_grade2       0
Ages[15-49]_All_grade3       0
Ages[20-29]_All_grade1       0
Ages[20-29]_All_grade2       0
Ages[20-29]_All_grade3       0
Ages[30-39]_All_grade1       1
Ages[30-39]_All_grade2       1
Ages[30-39]_All_grade3       1
Ages[40-49]_All_grade1       1
Ages[40-49]_All_grade2       1
Ages[40-49]_All_grade3       1
Ages[15-49]_Male_grade1      0
Ages[15-49]_Male_grade2      0
Ages[15-49]_Male_grade3      0
Ages[20-29]_Male_grade1      0
Ages[20-29]_Male_grade2      0
Ages[20-29]_Male_grade3      0
Ages[30-39]_Male_grade1      1
Ages[30-39]_Male_grade2      1
Ages[30-39]_Male_grade3      1
Ages[40-49]_Male_grade1      1
Ages[40-49]_Male_grade2      1
Ages[40-49]_Male_grade3      1
Ages[15-49]_Female_grade1    0
Ages[15-49]_Female_grade2    0
Ages[15-49]_Female_grade3    0
Ages[20-29]_Female_grade1    0
Ages[20-29]_Female_grade2    0
Ages[20-29]_Female_grade3    0
Ages[30-39]_Female_grade1    1
Ages[30-39]_Female_grade2    1
Ages[30-

In [13]:
if missing_values is not None and missing_values.any():
    # Create a report
    report = pd.DataFrame({
        'Column': missing_values.index,
        'Missing Values': missing_values.values,
        'Percentage Missing': (missing_values / len(df)) * 100
    })

    # Sort the report by the number of missing values in descending order
    report = report.sort_values(by='Missing Values', ascending=False)

    # Print the report
    print(report)
else:
    print("There are no missing values in the DataFrame.")

There are no missing values in the DataFrame.


# ***SUMMARY
Based on the ydata-profiling report, the dataset contains information on education attainment across different age groups, genders, and countries. The dataset consists of 38 columns and 626 rows, with no missing values present. The data includes information on the percentage of people in various age groups (15-49) who have completed different levels of education (grade 1 to grade 3), with separate columns for males and females. Additionally, the dataset covers a wide range of countries, including Afghanistan, Angola, and Albania, and spans a period of time from 1981 to 2020.

One of the key findings from the dataset is that the percentage of people who have completed higher levels of education (grade 3) tends to be lower than those who have completed lower levels of education (grade 1 or grade 2). This suggests that there may be barriers or challenges that prevent individuals from pursuing higher levels of education. Furthermore, the dataset reveals differences in education attainment between males and females, as well as between different countries. These differences may be influenced by a range of factors, including cultural and socio-economic factors.

Overall, the dataset provides valuable information on education attainment across different age groups, genders, and countries. The report generated by ydata-profiling helped to identify some interesting patterns and insights in the dataset, which can inform policy decisions and interventions aimed at improving education outcomes.***